scilpy/segment/recobundlesx.py

```diff
@@ -231,69 +229,65 @@ def _register_neighb_to_model(self, slr_num_thread=1,
             slm = slr.optimize(static, moving)
 
         if slr_transform_type_id >= 2:
-            init_transfo_dof = np.zeros(7)
-            init_transfo_dof[:6] = slm.xopt
-            init_transfo_dof[6] = 1.
-
-            slr = StreamlineLinearRegistration(metric=metric,
-                                               x0=init_transfo_dof,
-                                               bounds=bounds_dof[:7],
-                                               num_threads=slr_num_thread)
-            slm = slr.optimize(static, moving)
-
-        if slr_transform_type_id >= 3:
-            init_transfo_dof = np.zeros(9)
-            init_transfo_dof[:6] = slm.xopt[:6]
-            init_transfo_dof[6:] = np.array((slm.xopt[6],) * 3)
-
-            slr = StreamlineLinearRegistration(metric=metric,
-                                               x0=init_transfo_dof,
-                                               bounds=bounds_dof[:9],
-                                               num_threads=slr_num_thread)
-            slm = slr.optimize(static, moving)
-
-        return transform_streamlines(self.neighb_streamlines, slm.matrix)
-
-    def _prune_what_not_in_model(self, neighbors_to_prune,
+            if slr_transform_type_id == 2:
+                init_transfo_dof = np.zeros(7)
+                init_transfo_dof[:6] = slm.xopt
+                init_transfo_dof[6] = 1.
+
+                slr = StreamlineLinearRegistration(metric=metric,
+                                                   x0=init_transfo_dof,
+                                                   bounds=bounds_dof[:7],
+                                                   num_threads=slr_num_thread)
+                slm = slr.optimize(static, moving)
+
+            else:
+                init_transfo_dof = np.zeros(9)
+                init_transfo_dof[:6] = slm.xopt[:6]
+                init_transfo_dof[6:] = np.array((slm.xopt[6],) * 3)
+
+                slr = StreamlineLinearRegistration(metric=metric,
+                                                   x0=init_transfo_dof,
+                                                   bounds=bounds_dof[:9],
+                                                   num_threads=slr_num_thread)
+                slm = slr.optimize(static, moving)
+        self.model_centroids = transform_streamlines(self.model_centroids,
+                                                     np.linalg.inv(slm.matrix))
+
+    def _prune_what_not_in_model(self,
```

Comment:
Maybe `prune_outside_model` ? or `prune_far_from_model` ? The name `_prune_what_not_in_model` feels weird